src/app/country/services/country.service.ts

```diff
@@ -0,0 +1,21 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { environment } from '../../../environments/environment';
+import { Country } from '../../models/country';
+
+@Injectable()
+export class CountryService {
+  private countrySource = new BehaviorSubject<Country | null>(null);
+  selectedCountry$: any = this.countrySource.asObservable();
```

Comment:
I think it's better to keep this as private, and create a public method to expose it.